Doc/library/asyncio-sync.rst

```diff
@@ -1,305 +1,323 @@
+   statement::
 
-   When more than one coroutine is blocked in acquire() waiting for the state
-   to turn to unlocked, only one coroutine proceeds when a release() call
-   resets the state to unlocked; first coroutine which is blocked in acquire()
-   is being processed.
+       lock = asyncio.Lock()
 
-   :meth:`acquire` is a coroutine and should be called with ``await``.
+       # ... later
+       async with lock:
+           # access shared state
 
-   Locks support the :ref:`context management protocol <async-with-locks>`.
+   which is equivalent to::
 
-   This class is :ref:`not thread safe <asyncio-multithreading>`.
+       lock = asyncio.Lock()
 
-   .. method:: locked()
-
-      Return ``True`` if the lock is acquired.
+       # ... later
+       await lock.acquire()
+       try:
+           # access shared state
+       finally:
+           lock.release()
 
    .. coroutinemethod:: acquire()
 
-      Acquire a lock.
-
-      This method blocks until the lock is unlocked, then sets it to locked and
-      returns ``True``.
+      Acquire the lock.
 
-      This method is a :ref:`coroutine <coroutine>`.
+      This method waits until the lock is *unlocked*, sets it to
+      *locked* and returns ``True``.
 
    .. method:: release()
 
-      Release a lock.
+      Release the lock.
 
-      When the lock is locked, reset it to unlocked, and return.  If any other
-      coroutines are blocked waiting for the lock to become unlocked, allow
-      exactly one of them to proceed.
+      When the lock is *locked*, reset it to *unlocked* and return.
 
-      When invoked on an unlocked lock, a :exc:`RuntimeError` is raised.
+      If the lock is *unlocked* a :exc:`RuntimeError` is raised.
 
-      There is no return value.
+   .. method:: locked()
+
+      Return ``True`` if the lock is *locked*.
 
 
 Event
------
+=====
 
 .. class:: Event(\*, loop=None)
 
-   An Event implementation, asynchronous equivalent to :class:`threading.Event`.
+   An event object.  Not thread-safe.
 
-   Class implementing event objects. An event manages a flag that can be set to
-   true with the :meth:`set` method and reset to false with the :meth:`clear`
-   method.  The :meth:`wait` method blocks until the flag is true. The flag is
-   initially false.
+   An asyncio event can be used to notify multiple asyncio tasks
+   that some event has happened.
 
-   This class is :ref:`not thread safe <asyncio-multithreading>`.
+   An Event object manages an internal flag that can be set to *true*
+   with the :meth:`set` method and reset to *false* with the
+   :meth:`clear` method.  The :meth:`wait` method blocks until the
+   flag is set to *true*.  The flag is set to *false* initially.
 
-   .. method:: clear()
+   Example::
 
-      Reset the internal flag to false. Subsequently, coroutines calling
-      :meth:`wait` will block until :meth:`set` is called to set the internal
-      flag to true again.
+      async def waiter(e):
```

Comment:
Perhaps add a short docstring to the example: `e` is not defined for the reader until further down the page.

Reply:
I'll just give proper names to variables and arguments; using one-letter names is indeed very confusing.